### PR TITLE
feat: add virtiofs module check to image verify for OCI VM images

### PR DESCRIPTION
**Pull request by CMGS**
## Summary

- Add `oci.CheckInitramfsVirtiofs()` to scan kernel layer initramfs for `virtiofs.ko` module presence
- Supports gzip-compressed and uncompressed cpio newc archives; returns descriptive error for zstd
- Wire the check into `verifyOCILayoutBootability` so `cocoon image verify` reports virtiofs status
- Add `VirtiofsFound` / `VirtiofsChecked` fields to `BootCheckResult`
- Missing virtiofs module is reported as a warning (non-blocking), not an error

## Test plan

- [x] `TestCheckInitramfsVirtiofs_GzipWithVirtiofs` -- gzip cpio with virtiofs.ko returns found=true
- [x] `TestCheckInitramfsVirtiofs_GzipWithoutVirtiofs` -- gzip cpio without virtiofs returns found=false
- [x] `TestCheckInitramfsVirtiofs_UncompressedWithVirtiofsXZ` -- uncompressed cpio with virtiofs.ko.xz returns found=true
- [x] `TestCheckInitramfsVirtiofs_ZstdReturnsError` -- zstd magic bytes returns descriptive error
- [x] `TestCheckInitramfsVirtiofs_NoInitrdEntry` -- kernel tar without initrd returns found=false, no error
- [x] `make fmt` -- no formatting changes needed
- [x] `make lint` -- 0 issues (linux + darwin)
- [x] `make test` -- all packages pass with race detector

Closes docs/04.1 Section 6.5 requirement for initramfs virtiofs verification.

Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: oci/initramfs.go [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+		if rerr != nil {
+			return nil, fmt.Errorf("read kernel layer tar: %w", rerr)
+		}
+
+		// Normalize: strip leading "./" or "/".
+		name := strings.TrimPrefix(hdr.Name, "./")
+		name = strings.TrimPrefix(name, "/")
+
+		if _, ok := nameSet[name]; ok {
+			data, readErr := io.ReadAll(tr)
```
Comment (CMGS) [inline oci/initramfs.go]: suggestion: same note as the gzip path -- `io.ReadAll(tr)` on the tar entry has no size guard. A `LimitReader` wrapper here would provide symmetric protection. Non-blocking since the caller context is local verified blobs.

File: oci/initramfs.go [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+//	110     (namesize bytes, padded to 4-byte boundary)
+//	        (filesize bytes, padded to 4-byte boundary)
+func scanCpioForVirtiofs(data []byte) (bool, error) {
+	const headerLen = 110 // fixed cpio newc header size
+
+	offset := 0
+	for offset+headerLen <= len(data) {
+		// Verify magic.
+		magic := string(data[offset : offset+6])
+		if magic != "070701" {
```
Comment (CMGS) [inline oci/initramfs.go]: nit: when the magic doesn't match `070701`, the function silently breaks and returns `(false, nil)`. This is reasonable for trailing padding/garbage after TRAILER, but if the archive starts with a non-newc magic (e.g., old binary cpio format `070707`), it would silently report virtiofs-not-found rather than flagging an unsupported format. Consider adding a comment clarifying this is intentional, or returning an error when `offset == 0` and magic doesn't match.

File: oci/initramfs_test.go [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+	found, err := CheckInitramfsVirtiofs(tarPath)
+	if err != nil {
+		t.Fatalf("expected no error for missing initrd, got: %v", err)
+	}
+	if found {
+		t.Errorf("expected found=false when no initrd entry, got true")
+	}
+}
+
+func contains(s, substr string) bool {
```
Comment (CMGS) [inline oci/initramfs_test.go]: nit: the `contains()` helper reimplements `strings.Contains`. Since the test file already imports `bytes`, this works, but `strings.Contains(got, "zstd")` would be clearer and avoid the custom helper entirely.

File: oci/initramfs_test.go [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+		}
+	}
+
+	if err := tw.Close(); err != nil {
+		t.Fatalf("close tar: %v", err)
+	}
+	return tarPath
+}
+
+func TestCheckInitramfsVirtiofs_GzipWithVirtiofs(t *testing.T) {
```
Comment (CMGS) [inline oci/initramfs_test.go]: suggestion: all test cpio entries have zero-length file data (`nil` data arg). Consider adding one test case where at least one cpio entry has non-zero file data (e.g., a few hundred bytes). This would exercise the `dataTotal := alignUp(int(filesize), 4)` advancement in `scanCpioForVirtiofs` and confirm the parser correctly skips over file content. Currently that code path is technically untested.

**Review by CMGS — COMMENTED** [review]
LGTM with suggestions. Clean implementation -- cpio parser is correct, wiring is well-isolated, and test coverage is solid. A few non-blocking suggestions inline.
